hack/deployer/runner/tanzu.go

```diff
@@ -0,0 +1,420 @@
+		"AzureClientID":       t.azureCredentials.ClientID,
+		"AzureClientSecret":   t.azureCredentials.ClientSecret,
+		"AzureMachineType":    t.plan.MachineType,
+		"AzureLocation":       t.plan.Tanzu.Location,
+		"ResourceGroup":       t.plan.Tanzu.ResourceGroup,
+		"SSHPubKey":           t.plan.Tanzu.SSHPubKey,
+		"AzureSubscriptionID": t.azureCredentials.SubscriptionID,
+		"AzureTenantID":       t.azureCredentials.TenantID,
+		"NodeCount":           t.plan.Tanzu.NodeCount,
+	}
+	var cfg bytes.Buffer
+	if err := template.Must(template.New("tanzu-mgmt").Parse(TanzuInstallerConfigTemplate)).Execute(&cfg, params); err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(t.installerConfigFilePath(), cfg.Bytes(), 0600)
+}
+
+// installerConfigFilePath returns the path to the installer config valid in the context of deployer.
+func (t *TanzuDriver) installerConfigFilePath() string {
+	return filepath.Join(t.installerStateDir, TanzuInstallConfig)
+}
+
+// delete deletes the created resources simply by removing the Azure resource group
+func (t *TanzuDriver) delete() error {
+	return run([]func() error{
+		// shortcut deletion by not attempting deletion on application level but just delete the infrastructure
+		t.deleteResourceGroup,
+		t.deleteStorageContainer,
+		func() error {
+			user := fmt.Sprintf("%s-admin", t.plan.ClusterName)
+			context := fmt.Sprintf("%s@%s", user, t.plan.ClusterName)
+			return removeKubeconfig(context, t.plan.ClusterName, user)
+		},
+	})
+}
+
+// dockerizedTanzuCmd runs a tanzu CLI command inside a Docker container based off the Tanzu installer image.
+func (t *TanzuDriver) dockerizedTanzuCmd(args ...string) *Command {
+	params := map[string]interface{}{
+		"SharedVolume":        filepath.Join(SharedVolumeName(), filepath.Base(t.installerStateDir)),
+		"TanzuCLIDockerImage": t.plan.Tanzu.InstallerImage,
+		"Args":                args,
+	}
+	// We are mounting the shared volume into the installer container and configure it to be the HOME directory
+	// this is mainly so that we can save the installer state afterwards correctly as the Tanzu CLI is does not allow
+	// customizing where it saves its state. It always uses $HOME.
+	// We are mounting tmp as the installer needs a scratch space and writing into the container won't work
+	// We need host networking because the Tanzu installer will try to connect to a kind cluster it boots up
+	cmd := NewCommand(`docker run --rm \
+		-v {{.SharedVolume}}:/root \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v /tmp:/tmp \
+		-e HOME=/root \
+		-e PATH=/ \
+		--network host \
+		{{.TanzuCLIDockerImage}} \
+		/tanzu {{Join .Args " "}}`)
+	return cmd.AsTemplate(params)
+}
+
+// setup is a list of functions that need to run before any installer commands can run.
+func (t *TanzuDriver) setup() []func() error {
+	return []func() error{
+		t.loginToAzure,
+		t.loginToContainerRegistry,
+		t.ensureWorkDir,
+		t.ensureStorageContainer,
+		t.restoreInstallerState,
+		t.perpareCLI,
```

Comment:
It looks like we need to install the plugins (`prepareCLI()`) before we can call `az storage azcopy blob download` (`restoreInstallerState()`) otherwise the `az` CLI prompts you to install the storage-preview extension.
```
2021/09/20 09:57:47 Using installer state dir: /root/thb-dev-cluster3727324114   
2021/09/20 09:57:50 Reusing existing storage container to persist installer state
2021/09/20 09:57:50 Restoring installer state from storage container if any                                                                                                                                                         
exit status 2                                                                               
make: *** [Makefile:351: bootstrap-k8s] Error 1

# I manually ran the az command to get the error
root@fun:/go/src/github.com/elastic/cloud-on-k8s# az storage azcopy blob download -c ***** --account-name **** -s '*' -d "/root/*********"                     
The command requires the extension storage-preview. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n):
```
```suggestion
	return []func() error{
		t.loginToAzure,
		t.loginToContainerRegistry,
		t.ensureWorkDir,
		t.ensureStorageContainer,
		t.prepareCLI,
		t.restoreInstallerState,
```

Reply:
I'm completely off the mark, Tanzu CLI plugins != Azure CLI extensions :smile:

Reply:
We need to run `az extension add --name storage-preview`.

Reply:
Yes! I realised this as well. I started testing on CI today. Not sure why I did not hit it locally proabably because I already had installed the extension during development at some point. There is also an issue with the volume mounts when using a Docker volume. I will push additional commits.